### PR TITLE
Fix `deletetag` and `edittag` bugs

### LINKED ISSUE #163: Removing tags does not update the tag list in displayed problem list in UI
image

### PR DESCRIPTION
**Pull request by LuWenQ**
closes #163

### CODE REVIEW

File: src/main/java/seedu/algobase/logic/commands/tag/DeleteTagCommand.java [hunk 1]
```diff
@@ -45,6 +48,9 @@ public CommandResult execute(Model model, CommandHistory history) throws Command
         Tag tagToDelete = lastShownList.get(targetIndex.getZeroBased());
         model.deleteTags(tagToDelete);
         model.deleteTag(tagToDelete);
+        model.updateFilteredProblemList(PREDICATE_SHOW_ALL_PROBLEMS);
+        model.updateFilteredPlanList(Model.PREDICATE_SHOW_ALL_PLANS);
```
Comment (le0tan) [inline src/main/java/seedu/algobase/logic/commands/tag/DeleteTagCommand.java:52]: Why are these two lines changed?

File: src/main/java/seedu/algobase/logic/commands/tag/EditTagCommand.java [hunk 2]
```diff
@@ -64,7 +66,9 @@ public CommandResult execute(Model model, CommandHistory history) throws Command
         model.setTag(tagToEdit, editedTag);
         model.setTags(tagToEdit, editedTag);
 
-        model.updateFilteredTagList(PREDICATE_SHOW_ALL_TAGS);
+        model.updateFilteredProblemList(PREDICATE_SHOW_ALL_PROBLEMS);
+        model.updateFilteredPlanList(Model.PREDICATE_SHOW_ALL_PLANS);
+        model.updateFilteredTagList(Model.PREDICATE_SHOW_ALL_TAGS);
```
Comment (le0tan) [inline src/main/java/seedu/algobase/logic/commands/tag/EditTagCommand.java:71]: What happened here...?

**Review by le0tan — APPROVED** [review]
LGTM

File: src/main/java/seedu/algobase/logic/commands/tag/DeleteTagCommand.java (hunk 1 above)
Comment (LuWenQ) [reply src/main/java/seedu/algobase/logic/commands/tag/DeleteTagCommand.java:52]: Without these two lines the states of problems and plans will not refresh in the UI.

**Pull request merged**